bindata/etcd/cluster-restore.sh

```diff
@@ -61,9 +66,126 @@ function restore_static_pods() {
   done
 }
 
+function is_pacemaker_managed_etcd() {
+  if [ -n "${PACEMAKER_MANAGED_ETCD}" ]; then
+    return 0
+  fi
+
+  if systemctl status pacemaker >/dev/null 2>&1 ; then
+    PACEMAKER_MANAGED_ETCD=1
+    return 0
+  fi
+
+  return 1
+}
+
+function get_etcd_advertise_ip() {
+  # Try to detect IP from etcd.env (NODE_<nodename>_IP variable)
+  NODENAME_UNDERSCORE=$(echo "${NODENAME}" | tr '-' '_')
+  NODE_IP_VAR="NODE_${NODENAME_UNDERSCORE}_IP"
+  IP="${!NODE_IP_VAR}"
+  echo "$IP"
+}
+
+function wait_etcd_resources_started() {
+  local start=$SECONDS
+  local timeout=$((5*60))  # 5 minutes at most
+  while [ $((SECONDS - start)) -lt "$timeout" ]; do
+    if pcs status resources etcd-clone | grep "Started" | grep "master-0 master-1"  >/dev/null; then
+      return 0
+    fi
+    sleep 5
+  done
+
+  echo "timed out waiting for etcd resources to start (timeout: $timeout seconds)"
+  return 1
+}
+
+function ensure_etcd_container_stopped() {
+  echo "ensure ${ETCD_CONTAINER_NAME} container is stopped"
+  local start=$SECONDS
+  local timeout=120
+  local disable_etcd_called=0
+  while [ $((SECONDS - start)) -lt "$timeout" ]; do
+    if podman container exists "${ETCD_CONTAINER_NAME}" && \
+      [[ "$(podman inspect -f '{{.State.Status}}' "${ETCD_CONTAINER_NAME}")" != "exited" ]]; then
+
+      if [ "$disable_etcd_called" -eq 1 ]; then
+        sleep 1
+        continue
+      fi
+
+      if ! pcs resource disable etcd; then
+        echo "could not disable podman-etcd. Please ensure the ${ETCD_CONTAINER_NAME} container is stopped"
+        exit 1
+      fi
+
+      disable_etcd_called=1
+      continue
+    fi
+
+    echo "${ETCD_CONTAINER_NAME} container is stopped"
+    return 0
+  done
+
+  echo "timed out waiting for ${ETCD_CONTAINER_NAME} to stop"
+  return 1
+}
+
+function setup_pacemaker_restore() {
+  ETCD_ETCDCTL_RESTORE=1
+  ETCD_CONTAINER_NAME=${ETCD_CONTAINER_NAME:="etcd"}
+  PODMAN_ETCD_CONFIGURATION_FILES=(certs.hash config-previous.tar.gz config.yaml pod.yaml)
+
+  NODENAME=${NODENAME:-$(hostname)}
+  if [ -z "${NODENAME}" ] ; then
+    echo "could not determine the node hostname. Please set NODENAME env variable and try again"
+    exit 1
+  fi
+
+  IP=${ETCD_ADVERTISE_IP:-$(get_etcd_advertise_ip)}
+  if [ -z "${IP}" ]; then
+    echo "could not determine etcd advertise IP address from etcd.env. Please set ETCD_ADVERTISE_IP and try again"
+    exit 1
+  fi
+
+  # Extend restore flags with Pacemaker-specific arguments
+  ETCDCTL_RESTORE_FLAGS+=(
+    --skip-hash-check=true
```

Comment:
this is definitely something you should keep and not skip

Reply:
`podman-etcd` also makes copies (actually using `cp`) of the etcd db when a learner starts, which seemed to be incompatible with the hash check.
I was indeed considering to give some "special" name to such backups and let the script decide whether to use or not the hash check

Reply:
> which seemed to be incompatible with the hash check.

oof, that's probably something for us to take a deeper look at - I assume those are corrupted?